modules/cluster-logging-elasticsearch-retention.adoc

```diff
@@ -0,0 +1,102 @@
+= Configuring Elasticsearch retention policy
+
+
+Each Elasticsearch index adds load and overhead to the Elasticsearch cluster. You can use the Elasticsearch rollover index API to automatically rollover indices when they reach a specified age. When an index reaches the designated age, Elasticsearch rolls over the index, moving the current index and creating a new index. After seven days, the old index is automatically deleted. Rolling over and deleting indices optimizes the indices and helps Elasticsearch performance. 
+
+You specify how long Elasticsearch stores indices using a separate _retention policy_ for each of the three log sources, infrastucture logs, application logs, and audit logs, using the Cluster Logging Custom Resource (CR) to set the number of days after which the indices should be rolled over. The Cluster Logging CR writes the full policy to the Elasticsearch Custom Resource (CR).
+
+For example, the following Elasticsearch CR has a retention policy to roll over active indices for the infrastructure logs every three days and the old indices are deleted seven days after rollover. {product-title} checks every 15 minutes to determine if the infrastructure indices need to be rolled over.
+
+----
+apiVersion: "logging.openshift.io/v1"
+kind: "Elasticsearch"
+metadata:
+  name: "elasticsearch"
+spec:
+  indexManagement:
+    policies: <1>
+    - name: infra-policy
+      pollInterval: 15m
+      phases:
+        hot:
+          actions:
+            rollover:
+              maxAge: 3d
+        delete:
+          minAge: 7d
+    mappings: <2>
+    - name:  node.infra 
+      policyRef: infra-policy
+      aliases:
+      - infra  
+...
+----
+<1> For each log source, the retention policy indicates when to delete and rollover logs for that source. The policy includes: 
+* When to rollover the indices, determine by `maxAge`. This setting comes from the Cluster Logging CR.
+* When to check if rollover should be performed, determined by `pollInterval`. This setting is the default and cannot be changed.
+* When the rolled-over indicies will be deleted after the rollover, determined by `minAge`. This setting is the default and cannot be changed.
+<2> For each log source with a retention policy, the mapping connects the index, under alias, to the retention policy in `policyRef`.
+
+.Prerequisites
+
+* Cluster logging and Elasticsearch must be installed.
+
+.Procedure
+
+To configure the Elasticsearch retention policy:
+
+. Edit the Cluster Logging CR to add or modify the `retentionPolicy` parameter:
++
+----
+apiVersion: "logging.openshift.io/v1"
+kind: "ClusterLogging"
+...
+spec:
+  managementState: "Managed"
+  logStore:
+    type: "elasticsearch"
+    elasticsearch:
+      nodeCount: 2
+    retentionPolicy: <1>
+      application:
+        maxAge: 1d
+      infra:
+        maxAge: 7d
+      audit:
+        maxAge: 7d
+...
+----
+<1> Specify the maximum age that is allowed before triggering a rollover for each log source.
```

Comment:
The `maxAge` is for deleting, not for rollover.

Reply:
@QiaolingTang @jcantrill @sichvoge What are the defaults for the `maxAge` or does the user need to set a value for all three log types?

Reply:
@sichvoge @jcantrill @QiaolingTang  ^^

Reply:
@mburke5678, there is several conditions:
 1. when creating the clusterlogging instance, if users don't specify the retentionPolicy field, the CLO/EO(?) will create policies for all the 3 log types: http://pastebin.test.redhat.com/875892
2. if users only set retentionPolicy for one or two log types, then there will only have policy for the specified log type(s), and only the specified log type(s) logs could be received, details are in https://bugzilla.redhat.com/show_bug.cgi?id=1845788#c0 and https://bugzilla.redhat.com/show_bug.cgi?id=1845788#c4

Reply:
@QiaolingTang How does this look:
https://github.com/openshift/openshift-docs/pull/22403/commits/7fbda8283104fa1f931381d658718609cdde17fd